### DISCUSSION TITLE
Appium Connection Failure on CodeMagic: Unable to Run Tests

### DISCUSSION THREAD

**Discussion by vish7** [codemagic-ci-cd/codemagic-docs#1961] [Q&A]
<img width="1362" alt="image_2023_08_11T03_54_47_241Z" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/6234726/1a37adb4-7026-4101-a291-b01fafc7e51f">
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

1. Added a python script file in the root of the project to execute the test cases
2. Use the following command in the codemagic.yaml file 
 - name: Start Appium Server
      script: appium &
    - name: Check Appium Server Status
      script: curl http://localhost:4723/wd/hub/status
    - name: Run Appium Tests
      script: python ./LoginTest.py
3. Connection failure with Appium

### Expected results

Python file should be execute and run test cases

### Actual results

% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (7) Failed to connect to localhost port 4723 after 3 ms: Couldn't connect to server

### Build id (optional)

64d48b7d02786a66fb0d3519 , 64d383cc5314d192bbbabf79

**Comment by himesh-cm** [comment]
Hi @vish7, from the error `curl: (7) Failed to connect to localhost port 4723 after 3 ms: Couldn't connect to server` it seems appium is not running on the port `4723`, i suggest to [ssh](https://docs.codemagic.io/troubleshooting/accessing-builder-machine-via-ssh/) into the build machine and run those scripts manually, and make sure that you are using correct scripts.

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
